porcupine/plugins/rightclick_menu.py

```diff
@@ -0,0 +1,54 @@
+from __future__ import annotations
+
+import tkinter
+from typing import Callable, Dict, List
+
+from porcupine import get_main_window, get_tab_manager, tabs
+from porcupine.menubar import get_filetab
+
+_ftab_indexes: List[int] = []
+_menu_items: Dict[str, Callable[[tabs.FileTab], object] | Callable[[], object]] = {}
+
+
+def text_is_selected(tab: tabs.FileTab) -> bool:
+    try:
+        tab.textwidget.get("sel.first", "sel.last")
+    except tkinter.TclError:
+        # nothing selected
+        return False
+    return True
+
+
+def show_menu(event: tkinter.Event[tkinter.Misc]) -> None:
+    rm = tkinter.Menu(get_main_window(), tearoff=0)
+
+    for path, func in _menu_items.items():
+        rm.add_command(label=path, command=func)
+
+    flag = text_is_selected(get_filetab())
+    if not flag:
+        for i in _ftab_indexes:
+            rm.entryconfigure(i, state=tkinter.DISABLED)
+
+    rm.tk_popup(event.x_root, event.y_root)
+    rm.bind("<Unmap>", (lambda event: rm.after_idle(rm.destroy)), add=True)
+
+
+def add_rightclick_option(
+    path: str,
+    func: Callable[[tabs.FileTab], object] | Callable[[], object],
```

Comment:
I still don't think there is any need to support functions that don't take a `FileTab`.

Reply:
I edited the code myself to look more like how I imagined it. Let me know what you think.

Reply:
Ok, after thinking about it I now understand how functions that don't need the filetab parameter would 
ignore it.